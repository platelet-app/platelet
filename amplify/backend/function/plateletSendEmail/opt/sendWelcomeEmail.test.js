const aws = require("aws-sdk");
const { sendWelcomeEmail } = require("./sendWelcomeEmail");

jest.mock("aws-sdk", () => {
    const SES = jest.fn(() => ({
        sendEmail: jest.fn().mockReturnThis(),
        promise: jest.fn(),
    }));
    return { SES };
});

describe("sendWelcomeEmail", () => {
    it("should send a welcome email", async () => {
        process.env.REGION = "us-east-1";
        process.env.PLATELET_DOMAIN_NAME = "test.com";
        process.env.PLATELET_WELCOME_EMAIL = "welcome@test.com";
        const SESSpy = jest.spyOn(aws, "SES");
        const emailAddress = "test@test.com";
        const recipientName = "Test User";
        const password = "testpassword";
        await sendWelcomeEmail(emailAddress, recipientName, password);
        expect(SESSpy).toMatchSnapshot();
    });
});
