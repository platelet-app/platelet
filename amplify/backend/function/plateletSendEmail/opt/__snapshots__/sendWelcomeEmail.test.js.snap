// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sendWelcomeEmail should send a welcome email 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "apiVersion": "2010-12-01",
        "region": "us-east-1",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": {
        "promise": [MockFunction] {
          "calls": [
            [],
          ],
          "results": [
            {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "sendEmail": [MockFunction] {
          "calls": [
            [
              {
                "Destination": {
                  "ToAddresses": [
                    "test@test.com",
                  ],
                },
                "Message": {
                  "Body": {
                    "Html": {
                      "Charset": "UTF-8",
                      "Data": "
                    <p>
                        Welcome to https://test.com, Test User!
                    </p>
                    <p>
                        An admin has created your account for you with a temporary password.
                    </p>
                    <p>
                        <b>Username:</b> test@test.com
                    </p>
                    <p>
                        <b>Password:</b> testpassword
                    </p>
                    <p>
                        <b>This temporary password will expire in one week.</b>
                    </p>
                    <p>
                        Thank you.
                    </p>
                    ",
                    },
                    "Text": {
                      "Charset": "UTF-8",
                      "Data": "Welcome to https://test.com, Test User!
                    An admin has created your account for you. A temporary password has been generated for you.
                    Username: test@test.com
                    Password: testpassword
                    This temporary password will expire in one week.
                    Thank you.",
                    },
                  },
                  "Subject": {
                    "Charset": "UTF-8",
                    "Data": "Welcome to Platelet!",
                  },
                },
                "ReplyToAddresses": [
                  "welcome@test.com",
                ],
                "ReturnPath": "welcome@test.com",
                "Source": "welcome@test.com",
              },
            ],
          ],
          "results": [
            {
              "type": "return",
              "value": [Circular],
            },
          ],
        },
      },
    },
  ],
}
`;
